policy/reviewer/reviewer.go

```diff
@@ -0,0 +1,162 @@
+	randomTeam := allTeams[r.Intn(len(allTeams))]
+	teamInfo := strings.Split(randomTeam, "/")
+	teamMembers, err := prctx.ListTeamMembers(teamInfo[0], teamInfo[1])
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get member listing for team %s", randomTeam)
+	}
+	return teamMembers, nil
+}
+
+func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common.Result, r *rand.Rand) ([]string, error) {
+	logger := zerolog.Ctx(ctx)
+	pendingLeafNodes := findLeafChildren(result)
+	var requestedUsers []string
+
+	logger.Debug().Msgf("Collecting reviewers for %d pending leaf nodes", len(pendingLeafNodes))
+
+	for _, child := range pendingLeafNodes {
+		allUsers := make(map[string]struct{})
+		shoveIntoMap(allUsers, child.ReviewRequestRule.Users)
+
+		if len(child.ReviewRequestRule.Teams) > 0 {
+			teamMembers, err := selectTeamMembers(prctx, child.ReviewRequestRule.Teams, r)
+			if err != nil {
+				logger.Warn().Err(err).Msgf("Unable to get member listing for teams, skipping team member selection")
+			}
+			shoveIntoMap(allUsers, teamMembers)
+		}
+
+		if len(child.ReviewRequestRule.Organizations) > 0 {
+			randomOrg := child.ReviewRequestRule.Organizations[r.Intn(len(child.ReviewRequestRule.Organizations))]
+			orgMembers, err := prctx.ListOrganizationMembers(randomOrg)
+			if err != nil {
+				logger.Warn().Err(err).Msgf("Unable to get member listing for org %s, skipping org member selection", randomOrg)
+			}
+			shoveIntoMap(allUsers, orgMembers)
+		}
+
+		allCollaboratorPermissions, err := prctx.ListRepositoryCollaborators()
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to list repository collaborators")
+		}
+
+		if child.ReviewRequestRule.Admins {
+			var repoAdmins []string
+			for user, perm := range allCollaboratorPermissions {
+				if perm == common.GithubAdminPermission {
+					repoAdmins = append(repoAdmins, user)
+				}
+			}
+			shoveIntoMap(allUsers, repoAdmins)
+		}
+
+		if child.ReviewRequestRule.WriteCollaborators {
+			var repoCollaborators []string
+			for user, perm := range allCollaboratorPermissions {
+				if perm == common.GithubWritePermission {
+					repoCollaborators = append(repoCollaborators, user)
+				}
+			}
+			shoveIntoMap(allUsers, repoCollaborators)
+		}
+
+		var allUserList []string
+		for u := range allUsers {
+			// Remove any users who aren't collaborators or the author, since github will fail to assign _anyone_
+			// if the request contains one of these
+			_, ok := allCollaboratorPermissions[u]
+			if u != prctx.Author() && ok {
+				allUserList = append(allUserList, u)
+			}
+		}
+
+		logger.Debug().Msgf("Found %d total candidates for review after removing author and non-collaborators; randomly selecting %d", len(allUsers), child.ReviewRequestRule.RequiredCount)
+		randomSelection := selectRandomUsers(child.ReviewRequestRule.RequiredCount, allUserList, r)
+		requestedUsers = append(requestedUsers, randomSelection...)
```

Comment:
style: return `randomSelection` directly here and return `nil` at the end if no leaf nodes were found instead of declaring `requestedUsers`.

Reply:
We still want to iterate over all the leaf nodes, so we don't want to exit early.